mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -2734,6 +2734,113 @@ FailureOr<SmallVector<Value>> SoftmaxOp::decomposeOperation(OpBuilder &b) {
   return SmallVector<Value>{result};
 }
 
+//===----------------------------------------------------------------------===//
```

Comment:
These verifiers will not work for dynamic shapes. Can you support dynamic cases? The transform is only supported for static shapes right now, but shapes can become dynamic when tiling.

You can create an expected output shape from the input, allowing dynamic dims, and compare with the actual output shape. This helper may be useful: https://github.com/llvm/llvm-project/blob/586114510c5fa71d1377c7f53e68a3b12c472aa2/mlir/lib/IR/TypeUtilities.cpp#L58

This way will also make it easy to check that the batch/channel dimensions match for the input and output.

Reply:
Can we support static shapes in the upstream first? I added a TODO for dynamic cases.

Reply:
It has a tendency to not get done and not having the shape handling can mask other issues. Usually it's better to do it right, do it once.

Reply:
Thanks for your review. I updated the `verify()` functions to consider dynamic shapes. I also added test cases for dynamic shapes.